google/cloud/firestore_v1/stream_generator.py

```diff
@@ -38,3 +65,35 @@ def throw(self, exp=None):
 
     def close(self):
         return self._generator.close()
+    
+    @property
+    def explain_options(self) -> ExplainOptions:
+        """Query profiling options for this stream request."""
+        return self._explain_options
+
+    @property
+    def explain_metrics(self) -> ExplainMetrics:
+        """
+        Get the metrics associated with the query execution.
+        Metrics are only available when explain_options is set on the query. If
+        ExplainOptions.analyze is False, only plan_summary is available. If it is
+        True, execution_stats is also available.
+        :rtype: :class:`~google.cloud.firestore_v1.query_profile.ExplainMetrics`
+        :returns: The metrics associated with the query execution.
+        :raises: :class:`~google.cloud.firestore_v1.query_profile.QueryExplainError`
+            if explain_metrics is not available on the query.
+        """
+        if self._explain_metrics is not None:
+            return self._explain_metrics
+        elif self._explain_options is None:
+            raise QueryExplainError("explain_options not set on query.")
+        elif self._explain_options.analyze is False:
+            # we need to run the query to get the explain_metrics
+            try:
+                next(self)
+            except StopIteration:
+                pass
+            return self._explain_metrics
```

Comment:
there is a problem with the typing here: `self._explain_metrics` is optional, but this method says it always returns an `ExplainMetrics`. Maybe raise an exception if it's still None after calling next? Or is there a cleaner way?

Reply:
Indeed, I think it's better to add an exception if `explain_metrics` is empty. I'm a bit curious as to when this can happen, maybe the backend didn't send the proper data?

Reply:
done